driver-core/src/main/com/mongodb/internal/capi/MongoCryptHelper.java

```diff
@@ -138,6 +138,13 @@ private static BsonDocument toBsonDocument(final Map<String, Object> optionsMap)
         return new BsonDocumentWrapper<>(new Document(optionsMap), new DocumentCodec());
     }
 
+    public static boolean isMongocryptdSpawningDisabled(@Nullable final String cryptSharedLibVersion,
+            final AutoEncryptionSettings settings) {
+        boolean cryptSharedLibIsAvailable = cryptSharedLibVersion != null && !cryptSharedLibVersion.isEmpty();
+        boolean cryptSharedLibRequired = (boolean) settings.getExtraOptions().getOrDefault("cryptSharedLibRequired", false);
+        return settings.isBypassAutoEncryption() || settings.isBypassQueryAnalysis() || cryptSharedLibRequired || cryptSharedLibIsAvailable;
```

Comment:
I like that this is all in one place but wow what a return statement!!

Reply:
Yeah, I wish it were simpler...